### PR TITLE
Made convert tool use wildcard include exclude patterns

### PR DESCRIPTION
**Pull request by DreierF**
- [x] [No new Teamscale findings](https://build.cqse.eu/teamscale/delta.html#input/jacoco-client/?showMergeFindings=true)
- [ ] [Changes are tested adequately](https://build.cqse.eu/teamscale/tests.html#/jacoco-client/)
- [x] README.md updated in case of user-visible changes
- [x] CHANGELOG.md updated
- [x] build.gradle version number increased according to semantic versioning

### CODE REVIEW

File: report-generator/src/main/java/com/teamscale/report/testwise/jacoco/cache/ProbesCache.java [hunk 1]
```diff
@@ -59,14 +60,16 @@ public boolean containsClassId(long classId) {
...
-	public FileCoverageBuilder getCoverage(ExecutionData executionData) throws CoverageGenerationException {
+	public FileCoverageBuilder getCoverage(ExecutionData executionData, Predicate<String> locationIncludeFilter) throws CoverageGenerationException {
 		long classId = executionData.getId();
 		if (!containsClassId(classId)) {
-			logger.debug(
-					"Found coverage for a class " + executionData
-							.getName() + " that was not provided. Either you did not provide " +
-							"all relevant class files or you did not adjust the include/exclude filters on the agent to exclude " +
-							"coverage from irrelevant code.");
+			if (locationIncludeFilter.test(executionData.getName())) {
```
Comment (salsolatragus) [inline report-generator/src/main/java/com/teamscale/report/testwise/jacoco/cache/ProbesCache.java]: I take that if `!containsClassId() && !locationIncludeFilter.test()` then the filter and the cache are in sync and everything is alright, therefore we don't need to log anything, right?

File: agent/src/main/java/com/teamscale/jacoco/agent/convert/ConvertCommand.java [hunk 2]
```diff
@@ -48,7 +48,7 @@
 	 * Ant-style exclude patterns to apply during JaCoCo's traversal of class files.
```
Comment (salsolatragus) [inline agent/src/main/java/com/teamscale/jacoco/agent/convert/ConvertCommand.java]: Comment mentions Ant-style.

File: agent/src/main/java/com/teamscale/jacoco/agent/convert/ConvertCommand.java [hunk 3]
```diff
@@ -37,7 +37,7 @@
 	 * Ant-style include patterns to apply during JaCoCo's traversal of class files.
```
Comment (salsolatragus) [inline agent/src/main/java/com/teamscale/jacoco/agent/convert/ConvertCommand.java]: Comment mentions Ant-style.

File: agent/README.md [hunk 4]
```diff
@@ -37,6 +37,9 @@ The following options are available:
   For further details, please see the [JaCoCo documentation][jacoco-doc] in the "Agent" section.
 - `config-file` (optional): a file which contains one or more of the previously named options as `key=value` entries 
   which are separated by line breaks. The file may also contain comments starting with `#`. (For details see path format)
+- `logging-config`: path to a [logback][] configuration XML file (other configuration formats are not supported at the moment).
+  Use this to change the logging behaviour of the agent. Some sample configurations are provided with the agent in the
+  `logging` folder, e.g. to enable debug logging or log directly to the console. (For details see path format)
```
[the remark this answers is not in the document]
Comment (DreierF) [reply agent/README.md]: Was already there, someone just forgot to link it together and I overlooked it while copy-pasting 😉

File: agent/src/main/java/com/teamscale/jacoco/agent/convert/ConvertCommand.java (hunk 3 above)
Comment (DreierF) [reply agent/src/main/java/com/teamscale/jacoco/agent/convert/ConvertCommand.java]: 🙈

File: agent/README.md [hunk 5]
```diff
@@ -144,6 +144,8 @@ The agent's REST API has the following endpoints:
 - `[POST] /test/end/{uniformPath}` Signals to the agent that the test with the given uniformPath has just finished.
 - `[GET] /test` Returns the name of the current test. The result will be empty when the test already finished or was 
   not started yet.
+  
+The uniformPath parameter must be url encoded if it may contain special characters (including /).
```
[the remark this answers is not in the document]
Comment (DreierF) [reply agent/README.md]: Good point

[the remark this answers is not in the document]
Comment (DreierF) [reply agent/README.md]: Also renamed uniformPath as it's something a non TS-Dev doesn't know

File: agent/src/main/java/com/teamscale/jacoco/agent/testimpact/TestwiseCoverageAgent.java [hunk 6]
```diff
@@ -43,31 +43,46 @@ private void initServer() {
...
 	/** Handles the start of a new test case by setting the session ID. */
-	private void handleTestStart(Request request) {
-		logger.debug("Start test " + request.params(TEST_ID_PARAMETER));
+	private String handleTestStart(Request request, Response response) {
+		String testId = request.params(TEST_ID_PARAMETER);
+		if (testId == null || testId.isEmpty()) {
+			logger.error("End test name missing in " + request.url() + "!");
+
+			response.status(400);
+			response.body();
```
[the remark this answers is not in the document]
Comment (DreierF) [reply agent/src/main/java/com/teamscale/jacoco/agent/testimpact/TestwiseCoverageAgent.java]: Should not have been there ;-)

File: report-generator/src/main/java/com/teamscale/report/testwise/jacoco/cache/ProbesCache.java (hunk 1 above)
Comment (DreierF) [reply report-generator/src/main/java/com/teamscale/report/testwise/jacoco/cache/ProbesCache.java]: Exactly

**Comment by DreierF** [comment]
We now use ant patterns for all file patterns and wildcard patterns for packages+classfiles.
Therefore the remaining occurrences are ok AFAIKS.
Good catch! Changed it to README.md and adjusted the template.
No clue what's wrong with the build 🤔 Worked now again
RFR

**Comment by salsolatragus** [comment]
A side note: There doesn't seem to be coverage upload configured for this project. Should we remove the "adequately tested" item from the definition of done, because we cannot check it? Or should we configure coverage upload (probably the preferable way, I'm unaware of what this entails though).

**Pull request merged**